agent/engine/docker_task_engine_test.go

```diff
@@ -295,40 +312,41 @@ func TestRemoveEvents(t *testing.T) {
 			ExitCode: &exitCode,
 		},
 	}
-	steadyStateVerify <- time.Now()
 
 	if cont := <-contEvents; cont.Status != api.ContainerStopped {
 		t.Fatal("Expected container to stop first")
 		assert.Equal(t, *cont.ExitCode, 0, "Exit code should be present")
 	}
 	assert.Equal(t, (<-taskEvents).Status, api.TaskStopped, "Expected task to be STOPPED")
 
+	// Previous task is waiting for clean up, adding second task
 	sleepTaskStop := testdata.LoadTask("sleep5")
+	sleepTaskStop.Arn = "second_task"
 	sleepTaskStop.SetDesiredStatus(api.TaskStopped)
-
-	// Expect a bunch of steady state 'poll' describes when we warp 4 hours
-	client.EXPECT().DescribeContainer(gomock.Any()).AnyTimes()
-	client.EXPECT().RemoveContainer(gomock.Any(), gomock.Any()).Do(
-		func(removedContainerName string, timeout time.Duration) {
-			assert.Equal(t, createdContainerName, removedContainerName, "Container name mismatch")
-			// Emit a couple events for the task before the remove finishes; make sure this gets handled appropriately
-			eventStream <- createDockerEvent(api.ContainerStopped)
-			eventStream <- createDockerEvent(api.ContainerStopped)
-		}).Return(nil)
+	imageManager.EXPECT().RemoveContainerReferenceFromImageState(gomock.Any()).Return(nil)
```

Comment:
Can you add a comment here explaining why we should expect the `RemoveContainerReferenceFromImageState` call here?